fixed tooltips at tops of post lists

cellio opened the pull request:
Fixes https://github.com/codidact/qpixel/issues/1109 on:

- category page
- user posts page
- search-results page

There is a separate (pre-existing) placement problem on the search page, which I'll record separately.

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,16 +11,18 @@
-<div class="has-color-tertiary-500 category-meta" title="<%= post_count %>">
-  <div>
+<div class="has-color-tertiary-500 category-meta">
+  <div title="<%= post_count %> posts">
```
This div and the 2 that follow it (the post count, the Edit Category link, and the Subscribe button) need to be all contained within a single div. This time an extra layer of containment really is the answer. Without it, the styling of the next div up causes them to spread out to fill the available horizontal space (so the further you zoom out, the further they move apart from each other, rather than staying all on the left). I hadn't anticipated this when I suggested this would be an easy issue, but this will fix the spreading out problem.

cellio replied:
When I made these changes, the buttons all moved onto a new line.  I'm going to work through this again and see if I can figure out what's going on, but this didn't work on a first pass.

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,16 +11,18 @@
+<div class="has-color-tertiary-500 category-meta">
+  <div title="<%= post_count %> posts">
     <%= short_number_to_human post_count, precision: 1, significant: false %>
-    <%= 'post'.pluralize(post_count) %> &middot;
-
+    <%= 'post'.pluralize(post_count) %> 
+  </div>
```
Once the extra layer of containment has been added, this inner div around just the post count will cause it to be on a new line, rather than the same line as the Edit Category link and the Subscribe button.

This is because a div is a block element that defaults to being on its own line. If you change it to a span (which is an inline element) it will default to being on the same line as its neighbours.

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,16 +11,18 @@
+  </div>
+  <div>
     <% if current_user&.is_admin %>
       <%= link_to 'Edit Category', edit_category_path(@category) %>
       &middot;
-    <% end %>
-
+      <% end %>
+  </div>
```
The same problem happens with this div - it will default to being on its own line. It turns out it isn't needed so it can simply be removed which will solve the problem. Also a slight indentation problem - the `<% end %>` should be aligned with the `<% if ... %>` (and this whole Edit Category section will need one level of indentation less once its div is gone).

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,16 +11,18 @@
-
+      <% end %>
+  </div>
+  <div title="Get email notifications for new top-level posts">
```
I like the addition of title text here, but using a div is going to cause the same problem of defaulting to being on its own line. Instead the title can be added directly to the `<% link_to` element, and the div can be removed.

As this bit is Ruby rather than HTML, it will need to be `title:` rather than `title=`, and the style seems to be to use single quotes rather than double quotes for the Ruby sections. So it will end up looking something like this:

```ruby
    <%= link_to 'Subscribe',
                new_subscription_path(type: 'category', qualifier: @category.id, return_to: request.path),
                class: 'button is-outlined',
                title: 'Get email notifications for new top-level posts' %>
```

trichoplax reviewed (changes requested):
I like the addition of "posts" to the title text so it's no longer a bare number with no context.

There's a subtle problem I hadn't anticipated which only shows up when zoomed out or on a large screen. I've added some comments explaining how to fix it by adding a div and removing a couple of others.

I've only added comments to the Categories file for now - I wanted to get the feedback to you early. The other files may need similar changes if zooming out makes the buttons spread out. I'll check the other 2 files when I come back to this tomorrow.

cellio commented:
I've made some changes to the category page only (so far -- best to fully understand before doing the others).  This is what it currently looks like for an admin:

![screenshot](https://github.com/codidact/qpixel/assets/5557942/f2ccbfd8-c1ed-4a3a-8108-9876e4be698f)

and an anonymous user:

![screenshot](https://github.com/codidact/qpixel/assets/5557942/cc23bcda-44e3-4922-8b97-2b91302e334d)

This seems to do the right thing when zooming way out.  

I haven't seen the buttons flow to a second line, though some comments I haven't addressed yet suggest I should.  Can you review these changes and help me understand the unaddressed comments?  Thanks.

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,21 +11,23 @@
-    <%= short_number_to_human post_count, precision: 1, significant: false %>
-    <%= 'post'.pluralize(post_count) %> &middot;
+<div class="has-color-tertiary-500 category-meta">
+  <span>
```
This containing span is the main difference from my suggested approach. It doesn't seem to cause any problems so I see no reason not to stick with the span. (No change required here.)

trichoplax commented on app/views/categories/show.html.erb:
```diff
@@ -11,21 +11,23 @@
+  <span>
+    <span title="<%= post_count %> posts">
+      <%= short_number_to_human post_count, precision: 1, significant: false %>
+      <%= 'post'.pluralize(post_count) %> &nbsp;
```
The non breaking space here is a change from the middot. I don't have a preference between them myself, but just to clarify: The middot I suggested removing in the previous pull request was because there was no other button to separate the post count from, so it was serving no purpose. (No change required here - just explaining my reasoning.)

cellio replied:
Oh, I meant to comment on that.  I removed the middot because it was unnecessary.  Then things were too close together to my eye (the "edit" link was too close to the "N posts" text, in particular), so I added the spaces.

I never knew before this PR about the middot or that those stray specs I thought were monitor cruft or vision problems were intentional.  They seem too small to do anything other than make this reader wonder what's wrong. :-)   In this case we have "text - link - button", three different kinds of things; putting a tiny dot between the first two but not the second two would be inconsistent but putting one between the link and the button looks especially weird.  So I simplified.

trichoplax reviewed (approved):
Looks good - all 3 files are working well now.

I was only seeing the spreading out problem before an extra layer of containment was added. Your use of a span has solved this.

I was only seeing the items moving onto the next line when the inner divs were present. I can still recreate this in your version of the code by adding a div around the Edit Category link - it causes the 3 items to be arranged vertically rather than horizontally. Since you don't have that div anymore, there is no more problem.

I can't see any layout problems at different widths (such as mobile phone width) that weren't there already (the Random button doesn't quite fit on the same line as the other sorting buttons on a narrow mobile screen). This isn't relevant to this pull request so I'm happy to approve.
